File: Backend/TranscomaAPI/Comun/Entidades/Pedido.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace TranscomaAPI.Comun.Entidades
{
    public class Pedido : Entidad
    {
        private DateTime _fechaSolicitud;
        private DateTime _fechaEntrega;
        private string _estadoPedido;
        private string _destinatario;
        private Producto _producto;
        private Almacen _almacen;
        private Cliente _cliente;

        /// <summary>
        /// 
        /// </summary>
        /// <param name="id"></param>
        public Pedido(int id)
        {
            Id = id;
        }

        /// <summary>
        ///  Constructor para la consulta de un cliente en especifico
        /// </summary>
        /// <param name="id"></param>
        /// <param name="fechaSolicitud"></param>
        /// <param name="fechaEntrega"></param>
        /// <param name="estadoPedido"></param>
        /// <param name="destinatario"></param>
        /// <param name="producto"></param>
        /// <param name="almacen"></param>
        public Pedido(int id, DateTime fechaSolicitud, DateTime fechaEntrega, string estadoPedido, string destinatario, Producto producto, Almacen almacen)
        {
            Id = id;
            _fechaSolicitud = fechaSolicitud;
            _fechaEntrega = fechaEntrega;
            _estadoPedido = estadoPedido;
            _destinatario = destinatario;
            _producto = producto;
            _almacen = almacen;
        }

        /// <summary>
        ///  Contructor para la consulta de todas las pedidos
        /// </summary>
        /// <param name="id"></param>
        /// <param name="fechaSolicitud"></param>
        /// <param name="fechaEntrega"></param>
        /// <param name="estadoPedido"></param>
        /// <param name="destinatario"></param>
        /// <param name="producto"></param>
        /// <param name="almacen"></param>
        /// <param name="cliente"></param>
        public Pedido(int id, DateTime fechaSolicitud, DateTime fechaEntrega, string estadoPedido, string destinatario, Producto producto, Almacen almacen, Cliente cliente)
        {
            Id = id;
            _fechaSolicitud = fechaSolicitud;
            _fechaEntrega = fechaEntrega;
            _estadoPedido = estadoPedido;
            _destinatario = destinatario;
            _producto = producto;
            _almacen = almacen;
            _cliente = cliente;
        }

        /// <summary>
        /// 
        /// </summary>
        /// <param name="id"></param>
        /// <param name="fechaEntrada"></param>
        public Pedido(int id, DateTime fechaEntrada)
        {
            Id = id;
            _fechaSolicitud = fechaEntrada;
        }

        /// <summary>
        /// Getters y Setters
        /// </summary>
        public DateTime FechaSolicitud { get => _fechaSolicitud; set => _fechaSolicitud = value; }
        public DateTime FechaEntrega { get => _fechaEntrega; set => _fechaEntrega = value; }
        public string EstadoPedido { get => _estadoPedido; set => _estadoPedido = value; }
        public string Destino { get => _destinatario; set => _destinatario = value; }
        public Producto Producto { get => _producto; set => _producto = value; }
        public Almacen Almacen { get => _almacen; set => _almacen = value; }
        public Cliente Cliente { get => _cliente; set => _cliente = value; }
    }
}

File: frontend/src/app/models/Pedido.ts
import { EntidadBase } from "./EntidadBase";
import { Producto } from "./Producto";
import { Almacen } from "./Almacen";
import { Cliente } from "./Cliente";

export class Pedido extends EntidadBase {

    private _fechaSolicitud: Date;
    private _fechaEntrega: Date;
    private _estadoPedido: string;
    private _destinatario: string;
    private _producto: Producto;
    private _almacen: Almacen;
    private _cliente: Cliente;



    constructor(data?: any) {
        super();
        if (data !== undefined) {
            this.$id = data.id;
            this.$fechaSolicitud = data.fechaSolicitud;
            this.$fechaEntrega = data.fechaEntrega;
            this.$estadoPedido = data.estadoPedido;
            this.$destinatario = data.destinatario;
            if (data.producto !== undefined) {
                this.$producto = new Producto(data.producto);
            }
            else {
                this.$producto = null;
            }

            if (data.almacen !== undefined){
                this.$almacen = new Almacen(data.almacen);
            }
            else {
                this.$almacen = data.almacen;
            }

            if (data.cliente !== undefined) {
                this.$cliente = new Cliente(data.cliente);
            }
            else {
                this.$cliente = data.cliente;
            }
        }
    }

    /** Getters y setters */
    public get $fechaSolicitud(): Date {
        return this._fechaSolicitud;
    }

    public set $fechaSolicitud(value: Date) {
        this._fechaSolicitud = value;
    }

    public get $fechaEntrega(): Date {
        return this._fechaEntrega;
    }

    public set $fechaEntrega(value: Date) {
        this._fechaEntrega = value;
    }

    public get $estadoPedido(): string {
        return this._estadoPedido;
    }

    public set $estadoPedido(value: string) {
        this._estadoPedido = value;
    }

    public get $destinatario(): string {
        return this._destinatario;
    }

    public set $destinatario(value: string) {
        this._destinatario = value;
    }

    public get $producto(): Producto {
        return this._producto;
    }

    public set $producto(value: Producto) {
        this._producto = value;
    }

    public get $almacen(): Almacen {
        return this._almacen;
    }

    public set $almacen(value: Almacen) {
        this._almacen = value;
    }

    public get $cliente(): Cliente {
        return this._cliente;
    }

    public set $cliente(value: Cliente) {
        this._cliente = value;
    }
}